docs/development/env.md

```diff
@@ -0,0 +1,21 @@
+# Internal Rancher Desktop environment variables
```

Comment:
Not sure what this file has to do with the PR, but whatever... it looks correct.

Reply:
I wanted to document the new `RD_MOCK_MACOS_VERSION` variable, so I created this file for it. And so it didn't look too lonely I added some other recently referenced variables to it too.